### PR TITLE
Fixes stunbatons having one less charge than they should

### PR DESCRIPTION
**Pull request by PKPenguin321**
Makes the actual stun application into a proc, then uses that proc in the proc for stunning people. Code magic.

Fixes https://github.com/tgstation/-tg-station/issues/13209

### CODE REVIEW

**Comment by PKPenguin321** [comment]
welp i dont really know how to fix this but ill leave it up in case anybody wants to try to fix it

**Comment by Jordie0608** [comment]
Maybe try reading the travis log when it fails
`code/game/objects/items/weapons/stunbaton.dm:157:warning: L: unused label
code/game/objects/items/weapons/stunbaton.dm:162:warning: L: unused label`

File: code/game/objects/items/weapons/stunbaton.dm [hunk 1]
```diff
@@ -158,6 +144,24 @@
...
+/obj/item/weapon/melee/baton/proc/baton_stun(mob/living/L, mob/user)
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		if(H.check_shields(0, "[user]'s [name]", src, MELEE_ATTACK)) //No message; check_shields() handles that
+			playsound(L, 'sound/weapons/Genhit.ogg', 50, 1)
+			return 0
+	if(isrobot(loc))
+		var/mob/living/silicon/robot/R = loc
+		if(R || R.cell || R.cell.use(hitcost))
+			L.apply_stun
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/stunbaton.dm]: `L.apple_stun()`

File: code/game/objects/items/weapons/stunbaton.dm [hunk 2]
```diff
@@ -158,6 +144,24 @@
...
+		else
+			return 0
+	else
+		if(deductcharge(hitcost))
+			L.apply_stun
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/stunbaton.dm]: `L.apple_stun()`

**Comment by PKPenguin321** [comment]
well yo usee jordie i didn't realize that that was the problem i thought it was talking about the L not the () but thank you

File: code/game/objects/items/weapons/stunbaton.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/stunbaton.dm]: >apple_stun

**Comment by PKPenguin321** [comment]
bruh

**Comment by PKPenguin321** [comment]
ok new plan

**Pull request closed without merging**